src/models/Weather.tsx

```diff
@@ -0,0 +1,39 @@
+import WeatherDetails from './WeatherDetails';
+import MainWeatherDetails from './MainWeatherDetails';
+import WindDetails from './WindDetails';
+import SystemInformation from './SystemInformation';
+
+export default class Weather {
+
+    id: number;
+    coord: { lat: number, lon: number};
+    weather: WeatherDetails[];
+    base: string;
+    main: MainWeatherDetails;
+    wind: WindDetails;
+    clouds: { all: number };
```

Comment:
We also can extract the type here and avoid duplicated code on the constructor

Reply:
It seems that we don't need this value. Removed.